internal/types/base.go

```diff
@@ -0,0 +1,13 @@
+package types
+
+import (
+	"bytes"
+	"os"
+)
+
+// KeepalivedCollector is an interface for implementing multiple collector modes
+type KeepalivedCollector interface {
```

Comment:
I prefer a single `GetVRRP` method that abstracts all the signals and versions operations.

Reply:
The scenario for getting VRRP is same for all models so we will have duplicate codes if we have just `GetVRRP` method.

Reply:
Duplication Doesn't matter that much. From an architecture point of view, it doesn't make sense for the `collector` package to know about the internals of how to get data.
Its job should be getting the data and serving it.
The `Collector` implementations should handle the details on how to actually get the data from the keepalived.